OrbitGl/CaptureWindow.cpp

```diff
@@ -376,19 +376,23 @@ void CaptureWindow::MouseWheelMoved(int a_X, int a_Y, int a_Delta,
   if (delta > m_MaxWheelDelta) m_MaxWheelDelta = delta;
 
   float mousex = a_X;
+  float mousey = a_Y;
 
   float worldx;
   float worldy;
 
   ScreenToWorld(a_X, a_Y, worldx, worldy);
-  m_MouseRatio = static_cast<double>(mousex) / getWidth();
 
   bool zoomWidth = !a_Ctrl;
   if (zoomWidth) {
+    m_MouseRatio = static_cast<double>(mousex) / getWidth();
     time_graph_.ZoomTime(delta, m_MouseRatio);
     m_WheelMomentum = delta * m_WheelMomentum < 0 ? 0 : m_WheelMomentum + delta;
   } else {
-    // TODO: reimplement vertical zoom by scaling track heights.
+    m_MouseRatio = static_cast<double>(mousey) / getHeight();
+    time_graph_.VerticalZoom(delta, m_MouseRatio);
+    // TODO: What is m_WheelMomentun about?
+    m_WheelMomentum = delta * m_WheelMomentum < 0 ? 0 : m_WheelMomentum + delta;
```

Comment:
The wheel momentum code is not needed for vertical zoom, please remove.  It was added so that zoom would continue for a little while after the scroll wheel stopped moving. The effect was modulated by how fast the scroll wheel was moved. I don't think it works anymore actually.

Reply:
Perfect, thank you! 👍